sdk/typescript/src/cryptography/mnemonics.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import bip39 from '@scure/bip39';
+import { wordlist } from '@scure/bip39/wordlists/english';
+
+export function generateMnemonic(): string {
+  return bip39.generateMnemonic(wordlist);
+}
+
+/**
+ * Sanitize the mnemonics string provided by user
+ * @param mnemonics a 12-word string split by spaces that may contain mixed cases
+ * and extra spaces
+ * @returns a sanitized mnemonics string
+ */
+export function normalizeMnemonics(mnemonics: string): string {
+  return mnemonics
+    .trim()
+    .split(/\s+/)
+    .map((part) => part.toLowerCase())
+    .join(' ');
+}
+
+/**
+ * Parse and validate a path that is compliant to SLIP-0010 in form m/44'/784'/{account_index}'/{change_index}'/{address_index}'
+ *
+ * @param path path string (e.g. `m/44'/784'/0'/0'/0'`)
+ */
+export function isValidHardenedPath(path: string): boolean {
+  if (!new RegExp("^m\\/44'\\/784'\\/0'\\/[0-9]+'\\/[0-9]+'+$").test(path)) {
+    return false;
+  }
+  return true;
+}
+
+/**
+ * Parse and validate a path that is compliant to BIP-32 in form m/54'/784'/{account_index}'/{change_index}/{address_index}
+ * Note that the purpose for Secp256k1 is registered as 54, to differentiate from Ed25519 with purpose 44.
+ *
+ * @param path path string (e.g. `m/54'/784'/0'/0/0`)
+ */
+export function isValidBIP32Path(path: string): boolean {
+  if (!new RegExp("^m\\/54'\\/784'\\/[0-9]+'\\/[0-9]+\\/[0-9]+$").test(path)) {
+    return false;
+  }
+  return true;
+}
+
+/**
+ * Validate a 12-word mnemonic string in the English wordlist.
+ * 
+ * @param mnemonics 12 words string split by spaces
+ */
+export function validateMnemonics(mnemonics: string): boolean {
```

Comment:
I do wonder if we should be exporting utilities like these from the Sui SDK. Seems like maybe we could have some of the path derivation, but the mnemonics stuff seems a little more app-specific maybe? I imagine we'd want to allow non-english mnemonics in the future, and having to add support to the SDK methods seems non-ideal.

Reply:
I would argue the mnemonics -> privkey should live in the sdk, for other wallet builders to use as a standard instead of writing their own. In fact there are folks asking about the Secp256k1 keypair construction directly from the sdk. 

Also keeping two cryptography libs in wallet and sdk seem confusing.

Reply:
discussed offline, the wordlist import is now kept in /wallet whereas the derivation logic is in sdk/typescript